fix: Don't allow group chair to change group parent

pselkirk opened the pull request:
Fixes #6037

pselkirk committed:
fix: Don't allow group chair to change group parent (#6037)

rjsparks commented on ietf/group/tests_info.py:
```diff
@@ -950,6 +950,31 @@ def test_edit_description_field(self):
+        url = urlreverse('ietf.group.views.edit', kwargs={'acronym': group.acronym, 'action': 'edit', 'field': 'parent'})
+
+        # parent is not shown to group chair
+        login_testing_unauthorized(self, 'mars-chair', url)
+        r = self.client.get(url)
+        self.assertEqual(r.status_code, 302)
+        self.assertEqual(len(r.content), 0)
```
It's not clear what the intent of this test block is? The view that would only edit the parent field should probably just refuse entry?

What about testing the form that allows editing the whole group (when you don't pass in field to the view). - I would expect a test that shows that the parent field isn't present on that form.

pselkirk replied:
For the chair, /edit/parent/ behaves exactly like /edit/xyzzy/ - it renders an edit page with no editable fields, just a Submit button and a Back button, which both lead back to the about page.

However, I bolluxed the user creation, and what's actually going on there is a redirection to the login page. 🤦 

But yeah, I can test the whole group form.

pselkirk committed:
test: Fix test_edit_parent_field, add test_edit_parent (whole form)

rjsparks reviewed (changes requested):
Thanks for adjusting the tests. This is close to ready but has one more change that should be made.

Right now, in this branch, /group/rswg/edit/parent presents the following:

![image](https://github.com/ietf-tools/datatracker/assets/10996692/59b18324-e753-4e54-a8e3-0fef40e05f6e)

There is risk of unexpected happening with submitting a null-form to the current code (which hunts for fields).

I suspect the current code would also let someone provide the parent field manually here (or on the full edit form) in a post and the value would be honored.

Please test those posts (attempting to change parent as a chair should result in no change).

I think the form view at /group/rswg/edit/parent should 403 unless you have the roles you've already identified for showing the field.

pselkirk committed:
test: Verify that the chair can't circumvent the system to change the group parent

pselkirk commented:
It turns out that the form is accepted (form.is_valid==True), but form.cleaned_data only contains the fields that are shown to the user - in the case of editing just the parent field, it's an empty dict.

rjsparks commented:
I still think it's the right time to change the view behavior to 403 if there's no field to edit. (such as with your xyzzy example).

pselkirk committed:
fix: 403 if user tries to edit an unknown or hidden field

pselkirk commented:
Unit testing on github is failing with `'ietf/name/models.py:104: error: Need type annotation for "next_states"  [var-annotated]'`. Did something change in the settings?

pselkirk committed:
fix: Give edwg GroupFeatures a parent type

This tracks a change that was made directly in the production database
to fix the immediate cause of #6037.

pselkirk committed:
Empty commit to trigger github unit test
